test/threads_comprehensions.jl

```diff
@@ -0,0 +1,253 @@
+# This file is a part of Julia. License is MIT: https://julialang.org/license
+
+# Tests for @threads with array comprehensions
+using Test
+using Base.Threads
+
+@testset "@threads comprehensions" begin
+    # Test simple array comprehensions
+    @testset "simple comprehensions" begin
+        n = 1000
+        # Test default scheduling
+        result = @threads [i^2 for i in 1:n]
+        @test length(result) == n
+        @test all(result[i] == i^2 for i in 1:n)
+        @test issorted(result)  # should be ordered for default scheduling
+
+        # Test static scheduling
+        result_static = @threads :static [i^2 for i in 1:n]
+        @test length(result_static) == n
+        @test all(result_static[i] == i^2 for i in 1:n)
+        @test issorted(result_static)  # should be ordered for static scheduling
+
+        # Test dynamic scheduling
+        result_dynamic = @threads :dynamic [i^2 for i in 1:n]
+        @test length(result_dynamic) == n
+        @test all(result_dynamic[i] == i^2 for i in 1:n)
+        @test issorted(result_dynamic)  # should be ordered for dynamic scheduling
+
+        # Test greedy scheduling
+        result_greedy = @threads :greedy [i^2 for i in 1:n]
+        @test length(result_greedy) == n
+        @test all(result_greedy[i] == i^2 for i in 1:n)
+        @test issorted(result_greedy)  # should be ordered for greedy scheduling too
```

Comment:
These two tests seem wrong. `:greedy` does not guarantee any particular execution order, so there is no reason it should be preserved. It's better to check that every element in the result is also present when the same kind of collection is done without threading:


```suggestion
         result_nonthreaded = [ i^2 for i in 1:n ]
        @test all(result_greedy) do r
             r in result_nonthreaded
        end
```

This is of course quadratic, but since `:greedy` doesn't guarantee an order, anything else runs into ordering problems.